v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -170,7 +162,31 @@ func (c *Cleaner) migrateObjects(ctx context.Context, objectsToMigrate *unstruct
 			continue
 		}
 
-		err := retry.OnError(c.migrationBackoff, isErrorFatal, func() error { return c.client.Update(ctx, &obj) })
+		originalVersionFieldPath := []string{"spec", "originalVersion"}
+
+		originalVersion, found, err := unstructured.NestedString(obj.Object, originalVersionFieldPath...)
+		if err != nil {
+			return errors.Wrap(err,
+				fmt.Sprintf("error while migrating %q of kind %s", obj.GetName(), obj.GroupVersionKind().Kind))
```

Comment:
It's already an error, don't need to include that in the message.
```suggestion
				fmt.Sprintf("migrating %q of kind %s", obj.GetName(), obj.GroupVersionKind().Kind))
```